plugins/download_statistics/plugin_tests/download_statistics_test.py

```diff
@@ -0,0 +1,146 @@
+        self.assertEqual(data['downloadStatistics']['started'], count)
+        self.assertEqual(data['downloadStatistics']['requested'], count)
+        self.assertEqual(data['downloadStatistics']['completed'], count)
+
+    def testDownload(self):
+        # Create collection
+        collection = self.model('collection').createCollection('collection1', public=True)
+
+        # Create folder in collection
+        folder = self.model('folder').createFolder(collection, 'folder1',
+                                                   parentType='collection',
+                                                   public=True, )
+        # Create item
+        item = self.model('item').createItem('item1', self.admin, folder)
+
+        # Path to test files
+        file1Path = os.path.join(self.filesDir, 'txt1.txt')
+        file2Path = os.path.join(self.filesDir, 'txt2.txt')
+
+        file1 = ''
+        file2 = ''
+
+        # Upload files to item
+        with open(file1Path, 'rb') as fp:
+            file1 = self.model('upload').uploadFromFile(fp, os.path.getsize(file1Path),
+                                                        'txt1.txt', parentType='item',
+                                                        parent=item, user=self.admin)
+
+        with open(file2Path, 'rb') as fp:
+            file2 = self.model('upload').uploadFromFile(fp, os.path.getsize(file2Path),
+                                                        'txt2.txt', mimeType='image/jpeg',
+                                                        parentType='item', parent=item,
+                                                        user=self.admin)
+
+        # Download item and its files several times and ensure downloads are recorded
+        # Each file is downloaded 10 times
+        for n in range(0, 5):
+            self._downloadItem(item['_id'])
+            self._downloadFile(file1['_id'])
+            self._downloadFile(file2['_id'])
+
+        # Download entire collection
+        # Each file is downloaded 1 additional time
+        path = '/collection/%s/download' % collection['_id']
+        resp = self.request(path, user=self.admin, isJson=False)
+
+        # Iterate through generator to trigger download events
+        for data in resp.body:
+            data
+
+        # Download collection filtered by mime type
+        # file2 is downloaded one additional time
+        path = '/collection/%s/download' % collection['_id']
+        resp = self.request(path, user=self.admin, isJson=False, method='GET',
+                            params={
+                                'id': collection['_id'],
+                                'mimeFilter': json.dumps(['image/jpeg'])
+                            })
+
+        # iterate through generator to trigger download events
+        for data in resp.body:
+            data
+
+        self._checkDownloadsCount(file1['_id'], 11)
+        self._checkDownloadsCount(file2['_id'], 12)
```

Comment:
How hard would it be to test a download that's started but never completed? Or requested multiple times?